src/main/kotlin/org/rust/utils/formatterUtils.kt

```diff
@@ -0,0 +1,63 @@
+package org.rust.utils
+
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiWhiteSpace
+import org.rust.lang.core.psi.*
+
+fun RsNamedElement.getFormattedParts(): Pair<String, String> {
```

Comment:
That's a nice plan! Actually, I think tree elements can use some refactoring: I don't think we need separate tree classes for different elements.